texture.js

```diff
@@ -16,44 +16,53 @@ const FILTER = {
 export default class Texture {
     constructor(gl, image, wrapS, wrapT, filter, flipY) {
         const texture = gl.createTexture()
+        this.gl = gl
         this.texture = texture
+        this.image = image
+        this.flipY = flipY
+        this.wrapS = getProp(gl, WRAP, wrapS)
+        this.wrapT = getProp(gl, WRAP, wrapT)
+        this.filter = getProp(gl, FILTER, filter)
 
-        const ws = getProp(gl, WRAP, wrapS)
-        const wt = getProp(gl, WRAP, wrapT)
-        const f = getProp(gl, FILTER, filter)
-
-        if (image.complete && image.width && image.height) {
-            onImageReady(gl, texture, image, ws, wt, f, flipY)
+        if (image.complete && image.width && image.height || image instanceof Image === false) {
+            // Update if already loaded or not an image element
+            this.update()
         } else {
             // Fill texture with black pixel if image isn't ready
             gl.bindTexture(gl.TEXTURE_2D, texture)
             gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, new Uint8Array([0, 0, 0, 255]))
-            image.onload = function() {
-                image.onload = null
-                onImageReady(gl, texture, image, ws, wt, f, flipY)
-            }
+
+            image.addEventListener('load', function onload() {
```

Comment:
Memory leak - onload will never be unbinded because `onload` != `onload.bind(this)`. It's easier to use `image.onload` for this